.github/generate-quest-metadata.main.kts

```diff
@@ -0,0 +1,535 @@
+val sourceDirectory = projectDirectory.resolve("app/src/main/java/de/westnordost/streetcomplete/")
+val iconsDirectory = projectDirectory.resolve("res/graphics/quest icons/")
+
+val csvFile = projectDirectory.resolve("quest-list.csv")
+val jsonFile = projectDirectory.resolve("taginfo.json")
+
+val noteQuestName = "OsmNoteQuest"
+val noteQuestFile = sourceDirectory.resolve("data/osmnotes/notequests/OsmNoteQuestType.kt")
+
+val wikiRowSpan2 = " rowspan=\"2\" |"
+
+main(args)
+
+@Serializable
+data class Project(val name: String,
+                   val description: String,
+                   val project_url: String,
+                   val doc_url: String,
+                   val icon_url: String,
+                   val contact_name: String)
+
+@Serializable
+data class TaginfoTag(val key: String,
+                      val value: String,
+                      val description: String,
+                      val doc_url: String,
+                      val icon_url: String)
+
+data class TaginfoChange(val key: String,
+                         val value: String,
+                         val change: String)
+
+@Serializable
+data class Base(val data_url: String,
+                val project: Project,
+                val tags: List<TaginfoTag>,
+                val data_format: Int = 1)
+
+fun main(args: Array<String>) {
+    val task = if (args.contains("--task")) args[1 + args.indexOf("--task")] else "wiki"
+    println("Performing task: " + task)
+
+    val questFileContent = sourceDirectory.resolve("quests/QuestModule.kt").readText()
+    val questNameRegex = Regex("(?<=^ {8})[A-Z][a-zA-Z]+(?=\\()", RegexOption.MULTILINE)
+    val questNames = listOf(noteQuestName) + questNameRegex.findAll(questFileContent).map { it.value }
+
+    val questFiles = getFilesRecursively(sourceDirectory.resolve("quests/"))
+
+    when (task) {
+      "wiki" -> generateWikiCsv(questNames, questFiles)
+      "json" -> generateTaginfoJson(questNames, questFiles)
+      else -> { // Note the block
+        println("Unknown task " + task)
+      }
+    }
+}
+
+fun generateWikiCsv(questNames: List<String>, questFiles: List<File>) {
+    val strings = getStrings(projectDirectory.resolve("app/src/main/res/values/strings.xml"))
+    val wikiQuests = parseWikiTable(getWikiTableContent())
+    val repoQuests = questNames.mapIndexed { defaultPriority, name ->
+        getQuestWiki(name, defaultPriority, questFiles, strings, wikiQuests)
+    }.sortedBy { it.wikiOrder }
+
+    writeCsvFile(repoQuests, wikiQuests)
+}
+
+fun generateTaginfoJson(questNames: List<String>, questFiles: List<File>) {
+    val project = Project(
+        "StreetComplete",
+        "Surveyor app for Android",
+        "https://github.com/streetcomplete/StreetComplete",
+        "https://wiki.openstreetmap.org/wiki/StreetComplete",
+        "https://raw.githubusercontent.com/streetcomplete/StreetComplete/master/app/src/main/res/mipmap-xhdpi/ic_launcher.png",
+        "Peter Newman"
```

Comment:
Are you sure that you want to be contacted directly about maintaining this script?

Reply:
Possibly not, a bit of a placeholder at least. I think the spec suggested it should be an individual... :shrug: